File: convert/java/todo/src/main/java/dev/practice/todo/api/controller/response/TodoListResponse.java
package dev.practice.todo.api.controller.response;

import dev.practice.todo.domain.Todo;
import lombok.Builder;
import lombok.Getter;

import java.util.List;
import java.util.stream.Collectors;

@Getter
public class TodoListResponse {

    private List<TodoResponse> items;

    @Builder
    private TodoListResponse(List<TodoResponse> items) {
        this.items = items;
    }

    public static TodoListResponse of(List<Todo> todoList) {

        List<TodoResponse> todoListResponse = todoList.stream()
                .map(TodoResponse::of)
                .toList();

        return TodoListResponse.builder()
                .items(todoListResponse)
                .build();
    }
}

File: practice/convert/example1_todo/kotlin/todo/src/main/java/dev/practice/todo/api/controller/response/TodoListResponse.kt
package dev.practice.todo.api.controller.response

import dev.practice.todo.domain.Todo

data class TodoListResponse(
    val items: List<TodoResponse> //private 가 필요가 없는듯.. val 로 하여 setter 를 없앰
) {

    companion object{ // for static
        fun of(todoList: List<Todo>) =
            TodoListResponse(todoList.map(TodoResponse::of))
    }
}